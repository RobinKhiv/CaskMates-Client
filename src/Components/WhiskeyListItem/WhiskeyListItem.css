.WhiskeyList{
  border-radius: 2%;
 /* background-position: center; */
 background: linear-gradient(rgba(255,255,255,0.2), rgba(255,255,255,0.2)) center no-repeat;
  background-size: contain;
  background-color: white;
}
.listText{
    background-color: rgba(255, 255, 255, 0.575);
    border-radius: 1px;
    background-size: cover;

}
.WhiskeyList{
    border: 2px rgb(94, 54, 37) solid;
    margin-right: 10px;
    margin-bottom: 10px;
}
.whiskeyListName{
    font-size: 10px; 
    color: black;
}
.whiskeyListItem{
    text-decoration: none;
}
.ratingText{
    font-size: 12px;
    opacity: 1;
}
.row{
    height: 7em;
    display: flex;
    align-items: center;
    justify-content: center;
    flex-wrap: wrap;
}
.dropdown-add-btn{
    color: white;
    width: 45px;
    padding: 2px;
    font-size: 11px;
    border: none;
    cursor: pointer;
    text-align: center;
    font-weight: 400;
    background: rgb(94, 54, 37);
    border-radius: 0px;
    -moz-border-radius: 0px;
    -webkit-border-radius: 20px 20px 20px 20px;

  /* position: relative; */
  /* z-index:2; */
}
.add-selection{
    color: white;
    background: #21AA98;
    padding: 2px;
    font-size: 11px;
    border: none;
    cursor: pointer;
    text-align: center;
    /* font-weight: 900; */
    /* margin-bottom: 5px; */
 
    border-radius: 0px;
    -moz-border-radius: 0px;
    -webkit-border-radius: 20px 20px 20px 20px;
}
.add-selection, .triedLst{
    margin-bottom: 5px;
}
