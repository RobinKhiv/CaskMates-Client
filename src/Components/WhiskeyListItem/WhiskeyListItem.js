import React, { Component } from 'react';
import {Link} from 'react-router-dom';
import './WhiskeyListItem.css'
import WhiskeyApiService from '../../Services/whiskey-api-service';
import WhiskeyListContext from '../../Context/WhiskeyListContext';
import CheckedStar from '../StarRatings/CheckedStar';
import TokenService from '../../Services/token-api-service';


export default class WhiskeyListItem extends Component {
  constructor(){
    super();
    this.state={
      showAddMenu: false
    }
  }
  static contextType = WhiskeyListContext;

  showAddMenu = (e) => {
    e.preventDefault();
    this.setState({showAddMenu: true}, () => {
      document.addEventListener('click', this.closeAddMenu)
    });
  }
  
  closeAddMenu = e => {
    e.stopPropagation();
    this.setState({showAddMenu: false}, () => {
      document.removeEventListener('click', this.closeAddMenu)
    })
  }

  addButton = () =>{
    return (
    <input type="button" className="dropdown-add-btn" name="addWhiskey" value="Add" onClick={this.showAddMenu}/>
    )
  }

  addToList = (whiskey_Id, list_id) => {
    WhiskeyApiService.addToList(whiskey_Id, list_id)  
  }

  generateRatingStars= (value)=> {
     const checkedStar = [];
     if(value >= 1)
       for(let i = 0; i <= value; i++)
         checkedStar.push('star');
     return <CheckedStar checkedStar={checkedStar}/> 
  }

  render() {
    const {whiskey} = this.props;
    return (
    <div className="col-3 WhiskeyList" style={{backgroundImage: `url(${whiskey.image})`}} >
      <div className="listText">
      <Link to={`/whiskeys/${whiskey.whiskey_id}`} className='whiskeyListItem'>
        <div className="whiskeyListText">
          <h2 className='whiskeyListName'>{whiskey.whiskey_name}</h2>
          <p className="ratingText"> Rating: {this.generateRatingStars(whiskey.average_review_rating)} </p>
        </div>
      </Link>
      </div>
      <div className='dropdown-add'>
        {TokenService.hasAuthToken() ? this.addButton(): ''}
      {this.state.showAddMenu ? (
        <div className="dropdown-content" >
        <input type="button" name="FavoriteLst" className='add-selection' onClick={() => this.addToList(whiskey.whiskey_id, 1)} value="Favorite List"/><br/>
        <input type="button" name="wishLst" className='add-selection' onClick={() => this.addToList(whiskey.whiskey_id, 2)} value="Wish List"/><br/>
        <input type="button" name="alreadyTriedLst" className='add-selection triedLst' onClick={()=> this.addToList(whiskey.whiskey_id, 3)} value="Already Tried List"/><br/>
      </div>) : ""}
      </div>
    </div>
    )
  }
}



