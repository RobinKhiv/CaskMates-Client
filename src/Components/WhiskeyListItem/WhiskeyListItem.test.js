import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';
import WhiskeyListItem from './WhiskeyListItem';

describe('<WhiskeyListItem />', () => {
    const whiskey = {
        whiskey_id: 1,
        image: 'https',
        whiskey_name: 'name',
    };
    it('renders without crashing', () => { 
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <WhiskeyListItem whiskey={whiskey}/>
            </BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

  it('renders without crashing', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <WhiskeyListItem whiskey={whiskey}/>
        </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot(); 
  });
});