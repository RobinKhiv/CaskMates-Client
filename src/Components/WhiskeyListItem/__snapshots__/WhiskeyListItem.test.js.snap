// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WhiskeyListItem /> renders without crashing 1`] = `
<div
  className="col-3 WhiskeyList"
  style={
    Object {
      "backgroundImage": "url(https)",
    }
  }
>
  <div
    className="listText"
  >
    <a
      className="whiskeyListItem"
      href="/whiskeys/1"
      onClick={[Function]}
    >
      <div
        className="whiskeyListText"
      >
        <h2
          className="whiskeyListName"
        >
          name
        </h2>
        <p
          className="ratingText"
        >
           Rating: 
           
        </p>
      </div>
    </a>
  </div>
  <div
    className="dropdown-add"
  >
    
    
  </div>
</div>
`;
